kili/helpers.py

```diff
@@ -215,18 +216,34 @@ def fragment_builder(fields, type_of_fields):
     return fragment
 
 
-def deprecate(msg, _type=DeprecationWarning):
+def deprecate(
+        msg: Optional[str] = None,
+        removed_in: Optional[str] = None,
+        _type=DeprecationWarning):
     """
-    Decorator that generates a deprecation message
+    Decorator that tag a deprecated function.
+    - To deprecated the whole function, you can give a message at the decorator level.
+    - For more sharp condition on the warning message, integrate this warning inside the function
+    but still tag the function with this decorator, without giving a message argument
 
     Args:
-        msg: string message
+        msg: string message that will be displayed whenever the function is called
+        removed_in: string version in the format "Major.Minor"
+            in which the deprecation element has to be removed
         type: DeprecationWarning by default
     """
     def decorator(func):
```

Comment:
I think this decorator will obscure the type hint of the functions.

Reply:
The decorator can be typed with TypeVar  to enable type hints. https://stackoverflow.com/questions/54964760/is-this-decorator-type-annotated-correctly.